agent/handlers/v4/task_metadata_handler.go

```diff
@@ -78,6 +78,16 @@ func TaskMetadataHandler(state dockerstate.TaskEngineState, ecsClient api.ECSCli
 			taskResponse.Containers = responses
 		}
 
+		pulledContainers, ok := state.PulledContainerMapByArn(task.Arn)
+		// Append pulled containers to containers in task response
+		if ok {
```

Comment:
nit: no need to check for `ok' since you can iterate on a nil value: https://play.golang.org/p/CLoKMx0k4Fo

this could be simplified to:

```golang
pulledContainers, _ := state.PulledContainerMapByArn(task.Arn)
for _, dockerContainer := range pulledContainers {
	taskResponse.Containers = append(taskResponse.Containers, 
		NewPulledContainerResponse(dockerContainer, task.GetPrimaryENI()))
}
```

Reply:
ok is removed, and pulled container responses are directly appended to `taskResponse.Containers` as suggested. Thank you!